### PR TITLE
[BugFix] Instantiate 2 losses with different keys

### LINKED ISSUE #1545: [Bug] Cannot instantiate 2 losses with different keys
In MARL different agent groups have different loss instances.

It is not currently possible to instantiate the same loss 2 times with different keys

```python
loss_1 = DQNLoss(value_network=nn.Linear(3, 3), action_space="one_hot")
loss_1.set_keys(reward="a")
assert loss_1.tensor_keys.reward == "a"
loss_2 = DQNLoss(value_network=nn.Linear(3, 3), action_space="one_hot")
loss_2.set_keys(reward="b")
assert loss_1.tensor_keys.reward == "a"  # Fails, it is now b
```

### PR DESCRIPTION
**Pull request by matteobettini**
Fixes #1545

### CODE REVIEW

**Comment by matteobettini** [comment]
cc @Blonck

**Review by vmoens — APPROVED** [review]
LGTM

**Comment by vmoens** [comment]
image
For the record: I would tag this `FatFix`

**Comment by matteobettini** [comment]
ahahaha it is quite fat, lots of content

**Comment by matteobettini** [comment]
oh and it is happening only on gpu, let me try

**Comment by vmoens** [comment]
bug seems unrelated to this PR, merging
